clientv3/balancer_test.go

```diff
@@ -122,3 +130,88 @@ func TestBalancerGetBlocking(t *testing.T) {
 		t.Errorf("Get() with no up endpoints should timeout, got %v", err)
 	}
 }
+
+// TestBalancerDoNotBlockOnClose ensures that balancer and grpc don't deadlock each other
+// due to rapid open/close conn. The deadlock causes balancer.Close() to block forever.
+// See issue: https://github.com/coreos/etcd/issues/7283 for more detail.
+func TestBalancerDoNotBlockOnClose(t *testing.T) {
+	defer testutil.AfterTest(t)
+
+	kcl := newKillConnListener(t, 2)
+	defer kcl.close()
+
+	for i := 0; i < 100; i++ {
+		fmt.Printf("round %v \n", i)
+		sb := newSimpleBalancer(kcl.endpoints())
```

Comment:
Please just look at the test at https://github.com/heyitsanthony/etcd/blob/balancer-test/clientv3/balancer_test.go#L135. I took the time to make sure it actually deadlocks on master; what's here does not.